Update README.md, add architecture.md  and a new diagram (#110)
kyma-project/runtime-watcher: docs/04-10-architecture.md

To learn how to set up a listener, read the [Listener document](./Listener.md).

#### Subject Alternative Name (SAN) pinning

[SAN pinning](https://github.com/kyma-project/lifecycle-manager/blob/c1e06b7b973aca17cc715b6a4660b76f4e7b9e29/pkg/security/san_pinning.go#L55) is an example implementation of a validation function given to the listener package. SAN pinning is used in Lifecycle Manager. The validation function checks if the certificate of the incoming WatchEvent request has at least one matching SAN with the domain of the corresponding Kyma CR. The domain of a Kyma CR is saved in an annotation called `skr-domain`.

### Certificates

> Since it is required to have an mTLS connection from the Kyma cluster to the KCP Gateway, signed TLS certificates are needed for both the KCP Gateway and each Kyma cluster. In order not to rely on third parties in the infrastructure, we build the `Public Key Infrastructure`(PKI) by [bootstraping a self-signed CA issuer](https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers). The PKI uses the function of the [Cert-Manager](https://cert-manager.io/). 

Reviewer: - Can we switch to more active voice?
- No "we" according to our [style guide](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#pronouns).

```suggestion
Because you must have an mTLS connection from the Kyma cluster to the KCP Gateway,  you need signed TLS certificates for both the KCP Gateway and each Kyma cluster. To be independent from third parties in the infrastructure, Kyma built the `Public Key Infrastructure`(PKI) by [bootstraping a self-signed CA issuer](https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers). The PKI uses the function of the [Cert-Manager](https://cert-manager.io/). 
```
